src/plugins/shared/clusterOptions/selectClusterOptions.ts

```diff
@@ -0,0 +1,91 @@
+import { QuickPickItem, QuickPickItemKind, window } from "vscode";
+import { ReadyAzureSessionProvider } from "../../../auth/types";
+import { ClusterPreference } from "../types";
+import { Errorable } from "../../../commands/utils/errorable";
+import { RecentCluster } from "./state/recentCluster";
+import { selectExistingClusterOption } from "./options/selectExistingClusterOption";
+import { selectNewClusterOption } from "./options/selectNewClusterOption";
+import { selectRecentClusterOption } from "./options/selectRecentClusterOption";
+
+enum Options {
+    RecentCluster = "RecentCluster",
+    ExistingCluster = "ExistingCluster",
+    NewCluster = "NewCluster"
+}
+
+export type QuickPickClusterOptions = QuickPickItem & { type: Options }
+
+const getClusterOptions = () => {
+    const options = [
+        {
+            label: "", //seperator
+            type: Options.RecentCluster
+        },
+        {
+            label: "Recently used cluster",
+            type: Options.RecentCluster
+        },
+        {
+            label: "Existing cluster from your subscription",
+            type: Options.ExistingCluster
+        },
+        {
+            label: "Create new cluster from AKS VS Extension",
+            type: Options.NewCluster
+        }
+    ];
+
+    return options;
+}
+
+export async function selectClusterOptions(sessionProvider: ReadyAzureSessionProvider): Promise<Errorable<ClusterPreference | boolean>> {
```

Comment:
@kejatura-dev I don't see this used anywhere or called from extensions.ts. This needs to be exposed as command instead of plugins/shared? what do you think ? I am thinking might be useful if in future there are other usecases for extension to re-use this code piece, since we cannot hack much of tree view part. Thoughts? If you agree, this can be exposed as src/commands/aksSelectCluster so that when GH copilot invokes it would be vscode.commands.executeCommand("aksSelectCluster", args); Like you do for the other GH copilot command for create automatic cluster ?

Reply:
Thanks Hari for reviewing, I agree we should create a VS Code command for this. I will add the command a follow up PR